### DISCUSSION TITLE
Execute Implicit task Get step on the same worker

### DISCUSSION THREAD

**Discussion by Berber31** [concourse/concourse#7593] [Help & Support]
Is there a way  (option) to force all Implicit Get step on the same worker the task is executed ?

My container placement strategy is chaîned: limit-active-container, limit-active-volume, volume-locality, fewest-build-containers

But Implicit Get step (to pull image) and task exécution step are not executed on the same worker, so image pulled by Get step IS copied on the worker where the task IS executed.
And it takes longer than executing Implicit Get and executing task on the same worker.

**Comment by taylorsilva** [comment]
Using tags is one way or updating your placement strategy chain to prioritize `volume-locality` and possibly removing other strategies. There's no "switch" you can use to force them all on the same worker (besides [tags](https://concourse-ci.org/jobs.html#schema.step.tags)).

**Reply by jmccann** [reply]
I ran into this too ... seems like it'd make sense for the worker that is going to use the image to pull it itself regardless of strategies used or their order.  Should this be reported/tracked as a bug?

I'm on 7.4.3.

**Reply by jmccann** [reply]
So it seems like even if worker that eventually runs the task _already has the image_ that another worker will check/fetch the image first anyways.  But once task finally starts if worker running task already has image it starts right away.  If not then it "hangs" until the copy from other worker.

I also ran a combination of `volume-locality` and `limit-active-tasks` (with different numbers of allowed active tasks) and couldn't get anything to behave that made sense.  I even used just `volume-locality` and it still wouldn't use just a single worker.
